docs: Add some notes about dialogs (#585)
oamg/leapp: docs/source/dialogs.md

        ips = [entry[4][0] for entry in resolved if not '%' in entry[4][0]]
        self.produce(ResolvedHostname(name=hostname, ips=ips))
```

### Explaining the dialogs processing mechanism during the upgrade

> The upgrade itself, from the operator's point of view, consists of 3 distinct stages: preupgrade, remediate and upgrade.

Reviewer: ```suggestion
The upgrade itself, from the operator's point of view, consists of 3 distinct stages: preupgrade, remediation and upgrade.
```

Author: I meant for preupgrade\upgrade to be verbs in this context, so remediate makes more sense imho

Leapp preupgrade stage should be treated as "non-invasive system upgradeability analysis", when the upgrade workflow
stops right after preliminary system facts collection phases and a preupgrade report containing all the information
about potential issues is generated. If an actor containing dialog is discover during this stage, a specific
message is added to the preupgrade report file saying that for the successful upgrade the operator should record
their decision in the answerfile.

> As far as the dialogs are concerned, leapp remediate stage is intended specifically for answerfile management. The operator has the option to manually edit

Reviewer: ```suggestion
As far as the dialogs are concerned, remediation stage is intended specifically for answerfile management. The operator has the option to manually edit
```

Author: I meant for preupgrade\upgrade to be verbs in this context, so remediate makes more sense imho

        ips = [entry[4][0] for entry in resolved if not '%' in entry[4][0]]
        self.produce(ResolvedHostname(name=hostname, ips=ips))
```

> ### Explaining the dialogs processing mechanism during the upgrade

Reviewer 2: The title is suggesting something different from the text below. Basically, regarding the text below, it should be something like `Explaining the upgrade workflow`, if you know what I mean. In case it should be only about the dialogs, change text below, but I prefer to change the title.

> The upgrade itself from the operator's point of view consists of 3 distinct stages: leapp preupgrade, leapp answer

Reviewer: ```suggestion
The upgrade itself, from the operator's point of view, consists of 3 distinct stages: leapp preupgrade, leapp answer
```

> and leapp upgrade.

Reviewer 2: In such case, rather use: `preupgrade, remediation, upgrade`. using `leapp answer` is just part of remediation stage, during which user should do whatever else it is required regarding the report (e.g. create specific file, remove specific rpms, ....).

Leapp preupgrade stage should be treated as "non-invasive system upgradeability analysis", when the upgrade workflow
stops right after preliminary system facts collection phases and a preupgrade report containing all the information

> about potential issues is generated. If during this stage a dialog containing actor is discovered, a specific

Reviewer: ```suggestion
about potential issues is generated. If an actor containing dialog is discover during this stage, a specific
```

message is added to the preupgrade report file saying that for the successful upgrade the operator should record
their decision in the answerfile.

Leapp answer stage is intended specifically for answerfile management. The operator has the option to manually edit

> the answerfile with editor of choice or use leapp answer command to fill the answerfile (usually located at 

Reviewer: ```suggestion
the answerfile with editor of choice or use `leapp answer` command to fill the answerfile (usually located at 
```
